citar.el

```diff
@@ -921,14 +927,48 @@ With prefix, rebuild the cache before offering candidates."
 With prefix, rebuild the cache before offering candidates."
   (interactive (list (citar-select-refs
                       :rebuild-cache current-prefix-arg)))
-  (apply citar-insert-reference-function (list keys-entries)))
-
-(defun citar--insert-reference (keys-entries)
-  "Insert formatted reference(s) associated with the KEYS-ENTRIES."
-  (let ((template (citar-get-template 'preview)))
-    (dolist (key-entry keys-entries)
-      (when template
-        (insert (citar--format-entry-no-widths (cdr key-entry) template))))))
+  (insert (apply citar-format-reference-function (list keys-entries))))
+
+;;;###autoload
+(defun citar-copy-reference (keys-entries)
+  "Copy formatted reference(s) associated with the KEYS-ENTRIES.
+With prefix, rebuild the cache before offering candidates."
+  (interactive (list (citar-select-refs
+                      :rebuild-cache current-prefix-arg)))
+  (let ((references (apply citar-format-reference-function (list keys-entries))))
+    (if (not (equal "" references))
+        (progn
+          (kill-new references)
+          (message (format "Copied:\n  %s" references)))
+      (message "Key not found."))))
+
+(defun citar--format-reference (keys-entries)
```

Comment:
Minor thing: but why is this private, and the next public?

Reply:
Ah, that just carried over from citar--insert-reference. Changed to `citar-format-reference` in latest commit.